core/src/avm1/globals/blur_filter.rs

```diff
@@ -0,0 +1,201 @@
+//! flash.filter.BlurFilter object
+
+use crate::avm1::activation::Activation;
+use crate::avm1::error::Error;
+use crate::avm1::function::{Executable, FunctionObject};
+use crate::avm1::object::blur_filter::BlurFilterObject;
+use crate::avm1::{Object, TObject, Value};
+use enumset::EnumSet;
+use gc_arena::MutationContext;
+
+pub fn constructor<'gc>(
+    activation: &mut Activation<'_, 'gc, '_>,
+    this: Object<'gc>,
+    args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error<'gc>> {
+    let blur_x = args
+        .get(0)
+        .unwrap_or(&4.into())
+        .coerce_to_f64(activation)
+        .map(|x| x.max(0.0).min(255.0))?;
+
+    let blur_y = args
+        .get(1)
+        .unwrap_or(&4.into())
+        .coerce_to_f64(activation)
+        .map(|x| x.max(0.0).min(255.0))?;
+
+    let quality = args
+        .get(2)
+        .unwrap_or(&1.into())
+        .coerce_to_i32(activation)
+        .map(|x| x.max(0).min(15))?;
+
+    let blur_filter = this.as_blur_filter_object().unwrap();
+
+    blur_filter.set_blur_x(activation.context.gc_context, blur_x);
+    blur_filter.set_blur_y(activation.context.gc_context, blur_y);
+    blur_filter.set_quality(activation.context.gc_context, quality);
+
+    Ok(Value::Undefined)
+}
+
+pub fn clone<'gc>(
+    activation: &mut Activation<'_, 'gc, '_>,
+    this: Object<'gc>,
+    _args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error<'gc>> {
+    let proto = activation.context.avm1.prototypes.blur_filter_constructor;
```

Comment:
`clone` doesn't seem to be overridden in the filter subclasses. (Does it copy the underlying object instead?)
```
import flash.filters.*;
trace(BlurFilter.prototype.clone == BitmapFilter.prototype.clone);
var bf = new BlurFilter(10, 10);
var bf2 = BitmapFilter.prototype.clone.apply(bf);
trace(bf2.blurX);
```
Flash: true, 10
Ruffle: false, undefined